posts/2021-08-30-binding-app-to-postgresql-db-with-odo-and-ol-stack.adoc

```diff
@@ -0,0 +1,284 @@
+        name: sampledatabase
+      spec:
+        databaseCpu: 30m
+        databaseCpuLimit: 60m
+        databaseMemoryLimit: 512Mi
+        databaseMemoryRequest: 128Mi
+        databaseName: sampledb
+        databaseNameKeyEnvVar: POSTGRESQL_DATABASE
+        databasePassword: samplepwd
+        databasePasswordKeyEnvVar: POSTGRESQL_PASSWORD
+        databaseStorageRequest: 1Gi
+        databaseUser: sampleuser
+        databaseUserKeyEnvVar: POSTGRESQL_USER
+        image: centos/postgresql-96-centos7
+        size: 1
+  name: sampledatabase
+...
+----
+
+Push the updates to the cluster.
+
+[source,sh]
+----
+odo push
+----
+
+So far, two things have taken place:
+
+- The application was deployed on your cluster. The application was built and deployed using Maven. However, you could have also built and deployed the application using Gradle by using the `odo create java-openliberty-gradle mysboproj` command instead when creating the Java Open Liberty component.
+
+- A Dev4Ddevs Database resource instance was created. This, in turn, triggered the creation of a PostgreSQL database instance.
+
+However, the application is still not usable because it does not have the data needed to connect to the database. Let's solve that next.
+
+=== Bind the application to the PostgreSQL service
+
+List the available services to which the application can be bound. The PostgreSQL database service should be listed.
+
+[source,sh]
+----
+odo service list
+----
+
+Output:
+
+[source,sh]
+----
+NAME                        MANAGED BY ODO      STATE      AGE
+...
+Database/sampledatabase     Yes (mysboproj)     Pushed     50s
+...
+----
+
+Generate the service binding devfile configuration.
+
+[source,sh]
+----
+odo link Database/sampledatabase
+----
+
+Configuration added to devfile.yaml:
+
+[source,sh]
+----
+...
+- kubernetes:
+    inlined: |
+      apiVersion: binding.operators.coreos.com/v1alpha1
+      kind: ServiceBinding
+      metadata:
+        creationTimestamp: null
+        name: mysboproj-database-sampledatabase
+      spec:
+        application:
+          group: apps
+          name: mysboproj-app
+          resource: deployments
+          version: v1
+        bindAsFiles: false
+        detectBindingResources: true
+        services:
+        - group: postgresql.dev4devs.com
+          kind: Database
+          name: sampledatabase
+          version: v1alpha1
+      status:
+        secret: ""
+  name: mysboproj-database-sampledatabase
+...
+----
+
+Push the updates to the cluster. 
+
+[source,sh]
+----
+odo push
+----
+
+That is all. The application is now bound to the PostgreSQL database service. During the binding process, a secret containing the database connection information was created, and the pod hosting the application is restarted with the database connection information contained in the secret. The connection information is set in the application pod as environment variables.
+
+Let’s make sure we can use the application next.
```

Comment:
Let’s make sure we can use the application next.
--->
Next, let’s make sure we can use the application.